chunky/src/java/se/llbit/chunky/world/Chunk.java

```diff
@@ -206,13 +299,13 @@ private void loadBiomes(@NotNull Tag data, ChunkData chunkData) {
    * @param biomesTag the .Level.Biomes NBT tag to load data from.
    * @param output a byte array of length 16x16.
    */
-  private void extractBiomeData(@NotNull Tag biomesTag, ChunkData output) {
+  private void extractBiomeDataPre21w39a(@NotNull Tag biomesTag, ChunkData output, BiomePalette biomePalette) {
```

Comment:
Maybe in the future this could be refactored to some `BiomeExtractor` interface? But that's for another time. :upside_down_face:

Reply:
Yeah, there is a lot of code that needs to be moved out of `Chunk` at this point...